### DISCUSSION TITLE
Place my custom card in room NOT in a sub page

### DISCUSSION THREAD

**Discussion by konradwalsh** [dwainscheeren/dwains-lovelace-dashboard#267] [Q&A] [answered]
How do I place my card in a room?

I have an 'Outdoors' room

I have a custom 'flip' card for multiple actions.

I want to click on my Outdoors Room and see my flip card. I dont want to click Outdoors/Sub Page then see it.

![image](https://user-images.githubusercontent.com/7996482/106724699-70986200-6600-11eb-95bd-357f680741d6.png)

How is this achieve?

**Comment by dwainscheeren** [comment, marked as answer]
So you want to add a own card on the room page?

**Reply by konradwalsh** [reply]
Yep, if possible to save an extra click/page load

**Reply by dwainscheeren** [reply]
If you create a room addon you have a page.yaml and button.yaml. You can place the code of the page.yaml inside the button.yaml, its sort of a hack. Then the code of your page is displayed on the place of the button so directly in the room overview.

**Reply by konradwalsh** [reply]
Ah.. great.. ok.. I will read to docs on that and then post my solution here for anyone else trying

**Reply by konradwalsh** [reply]
![image](https://user-images.githubusercontent.com/7996482/106737945-3d110400-660f-11eb-9525-89550ea7fe34.png)

I'm gonna need a pointer on this

I followed the guide... but the example hello-room isnt part of 2.0
So I copied over the version from 1.5 and copied in the code from the docs


But this is what the room looks like after reload/ctrl-f5

![image](https://user-images.githubusercontent.com/7996482/106738312-ad1f8a00-660f-11eb-918b-c200a66279f4.png)

**Reply by dwainscheeren** [reply]
You need to have a button there then now from the example. Check your files and button.yaml

**Reply by konradwalsh** [reply]
im sorry.. but I need more help on this..

This is the code
![image](https://user-images.githubusercontent.com/7996482/106741707-f7a30580-6613-11eb-886e-ed8af0436468.png)

`  - name: Back Garden
    icon: mdi:flower
    light: light.back_garden_lights
    door: binary_sensor.french_door_status
    addons:
      - name: Hello room
        icon: fas:puzzle-piece
        path: "dwains-dashboard/addons/rooms/hello-room/page.yaml"
        button_path: "dwains-dashboard/addons/outdoors/outdoors-card.yaml"
        data:
          entity: binary_sensor.french_door_status`

I put in my code for the button_path:

` button_path: "dwains-dashboard/addons/outdoors/outdoors-card.yaml"`


This should load this:
![image](https://user-images.githubusercontent.com/7996482/106741601-d04c3880-6613-11eb-9555-eae0042d85b5.png)

**Reply by dwainscheeren** [reply]
What is the content of `outdoors-card.yaml?`

**Reply by konradwalsh** [reply]
`# dwains_theme

- type: custom:dwains-flexbox-card
  items_classes: "col-xs-12 col-md-12 col-lg-6 col-xl-6"
  cards:
    - type: vertical-stack
      cards:
        - type: entities
          style: |
            ha-card {
              border-radius: 30px;
              background: rgba(0,0,0,.3);
              box-shadow: none;
              }
          entities:
            - type: "custom:paper-buttons-row"
              buttons:
                - entity: input_select.lighttoggle
                  name: false
                  icon: "hass:sofa-single"
                  state_styles:
                    Boudoir:
                      button:
                        background: "rgba(255,255,255,.5)"
                        box-shadow: "0 0 15px rgba(255,255,255,.6)"
                  style:
                    button:
                      - background: "rgba(255,255,255,.1)"
                      - border-radius: 25px
                      - padding: 20px
                    icon:
                      - "--mdc-icon-size": 40px
                  tap_action:
                    action: call-service
                    service: input_select.select_option
                    service_data:
                      entity_id: input_select.lighttoggle
                      option: Boudoir
                - entity: input_select.lighttoggle
                  name: false
                  icon: "hass:bed-king"
                  state_styles:
                    Bedroom:
                      button:
                        background: "rgba(255,255,255,.5)"
                        box-shadow: "0 0 15px rgba(255,255,255,.6)"
                  style:
                    button:
                      - background: "rgba(255,255,255,.1)"
                      - border-radius: 25px
                      - padding: 20px
                    icon:
                      - "--mdc-icon-size": 40px
                  tap_action:
                    action: call-service
                    service: input_select.select_option
                    service_data:
                      entity_id: input_select.lighttoggle
                      option: Bedroom
                - entity: input_select.lighttoggle
                  name: false
                  icon: "hass:laptop"
                  state_styles:
                    Office:
                      button:
                        background: "rgba(255,255,255,.5)"
                        box-shadow: "0 0 15px rgba(255,255,255,.6)"
                  style:
                    button:
                      - background: "rgba(255,255,255,.1)"
                      - border-radius: 25px
                      - padding: 20px
                    icon:
                      - "--mdc-icon-size": 40px
                  tap_action:
                    action: call-service
                    service: input_select.select_option
                    service_data:
                      entity_id: input_select.lighttoggle
                      option: Office
                - entity: input_select.lighttoggle
                  name: false
                  icon: "hass:fridge"
                  state_styles:
                    Kitchen:
                      button:
                        background: "rgba(255,255,255,.5)"
                        box-shadow: "0 0 15px rgba(255,255,255,.6)"
                  style:
                    button:
                      - background: "rgba(255,255,255,.1)"
                      - border-radius: 25px
                      - padding: 20px
                    icon:
                      - "--mdc-icon-size": 40px
                  tap_action:
                    action: call-service
                    service: input_select.select_option
                    service_data:
                      entity_id: input_select.lighttoggle
                      option: Kitchen
        - type: "custom:state-switch"
          entity: input_select.lighttoggle
          default: Boudoir
          transition: flip
          states:
            Boudoir:
              type: "custom:mod-card"
              style: |
                ha-card {
                  background: rgba(0,0,0,.3);
                  padding: 10px;
                  border-radius: 35px;
                  margin-top: none;
                  }
              card:
                type: grid
                columns: 3
                cards:
                  - type: "custom:button-card"
                    icon: "hass:desk-lamp"
                    aspect_ratio: 1/1
                    tap_action:
                      action: toggle
                    entity: light.desk_lamp
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          icon:
                            - color: var(--button-card-light-color)
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    entity: light.standing_lamp
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: var(--button-card-light-color)
                            - animation: spin 2s linear infinite
                            - background: >-
                                radial-gradient(circle, rgba(255,198,0,1) 0%,
                                rgba(255,255,255,1) 54%, rgba(0,0,0,1) 100%);
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    entity: group.boudoir
                    icon: "hass:ceiling-light"
                    name: Ceiling Lights
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                        - color: white
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: var(--button-card-light-color)
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    entity: switch.air_purifier
                    icon: "hass:fan"
                    aspect_ratio: 1/1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px .75em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        spin: true
                        styles:
                          icon:
                            - color: "rgb(1, 182, 253)"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                    tap_action:
                      action: toggle
                  - type: "custom:button-card"
                    entity: light.nanoleaf
                    aspect_ratio: 1/1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                      name:
                        - padding: 8px
                        - font-family: Roboto
                        - font-size: 20px
                    tap_action:
                      action: toggle
                    name: Nanos
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          icon:
                            - color: var(--button-card-light-color)
                          name:
                            - color: "rgba(0,0,0,.6)"
                            - font-weight: bold
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    icon: "hass:television"
                    name: TV
                    aspect_ratio: 1/1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      state:
                        - color: "rgba(255,255,255,1)"
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - font-family: DM Sans
                        - background: "rgba(255,255,255,.1"
                    tap_action:
                      action: more-info
                    entity: media_player.b_chromecast
                    show_state: true
                    state:
                      - value: playing
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          state:
                            - animation: blink 2s ease infinite
                            - color: "rgb(100,100,255)"
                          icon:
                            - color: "rgb(100,100,255)"
                          name:
                            - color: "rgba(0,0,0,.6)"
                            - font-weight: bold
            Kitchen:
              type: "custom:mod-card"
              style: |
                ha-card {
                  background: rgba(0,0,0,.3);
                  padding: 10px;
                  border-radius: 35px;
                  margin-right: 10px;
                  margin-top: none;
                  }
              card:
                type: grid
                columns: 3
                cards:
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    tap_action:
                      action: toggle
                    entity: group.kitchen
                    name: All Lights
                    icon: "hass:ceiling-light"
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          icon:
                            - color: var(--button-card-light-color)
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    icon: "mdi:outdoor-lamp"
                    entity: switch.hallway_lamp
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: "rgb(255, 178, 0)"
                            - animation: spin 2s linear infinite
                            - background: >-
                                radial-gradient(circle, rgba(255,198,0,1) 0%,
                                rgba(255,255,255,1) 54%, rgba(0,0,0,1) 100%);
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      state:
                        - color: "rgba(255,255,255,1)"
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - font-family: DM Sans
                        - background: "rgba(255,255,255,.1"
                    tap_action:
                      action: more-info
                    entity: vacuum.rug_b
                    show_state: true
                    state:
                      - value: cleaning
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          state:
                            - animation: blink 2s ease infinite
                            - color: "rgb(100,100,255)"
                          icon:
                            - color: "rgb(100,100,255)"
                          name:
                            - color: "rgba(0,0,0,.6)"
            Bedroom:
              type: "custom:mod-card"
              style: |
                ha-card {
                  background: rgba(0,0,0,.3);
                  padding: 10px;
                  border-radius: 35px;
                  margin-right: 10px;
                  margin-top: none;
                  }
              card:
                type: grid
                columns: 3
                cards:
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    tap_action:
                      action: toggle
                    entity: switch.vine_lights
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          icon:
                            - color: var(--button-card-light-color)
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    icon: "mdi:floor-lamp"
                    entity: switch.lamp
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: "rgb(255, 178, 0)"
                            - animation: spin 2s linear infinite
                            - background: >-
                                radial-gradient(circle, rgba(255,198,0,1) 0%,
                                rgba(255,255,255,1) 54%, rgba(0,0,0,1) 100%);
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    icon: "hass:television"
                    name: TV
                    aspect_ratio: 1/1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      state:
                        - color: "rgba(255,255,255,1)"
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - font-family: DM Sans
                        - background: "rgba(255,255,255,.1"
                    tap_action:
                      action: toggle
                    entity: media_player.den_tv
                    show_state: true
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          state:
                            - animation: blink 2s ease infinite
                            - color: "rgb(100,100,255)"
                          icon:
                            - color: "rgb(100,100,255)"
                          name:
                            - color: "rgba(0,0,0,.6)"
                            - font-weight: bold
            Office:
              type: "custom:mod-card"
              style: |
                ha-card {
                  background: rgba(0,0,0,.3);
                  padding: 10px;
                  border-radius: 35px;
                  margin-right: 10px;
                  margin-top: none;
                  }
              card:
                type: grid
                columns: 3
                cards:
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    tap_action:
                      action: toggle
                    entity: light.office_strip
                    icon: "hass:led-strip-variant"
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          icon:
                            - color: var(--button-card-light-color)
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    icon: "mdi:desk-lamp"
                    entity: light.desk_lamp_1
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: "rgb(255, 178, 0)"
                            - animation: spin 2s linear infinite
                            - background: >-
                                radial-gradient(circle, rgba(255,198,0,1) 0%,
                                rgba(255,255,255,1) 54%, rgba(0,0,0,1) 100%);
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
                  - type: "custom:button-card"
                    aspect_ratio: 1/1
                    icon: "mdi:desk-lamp"
                    entity: light.desk_lamp_2
                    style: |
                      ha-card:hover {
                        transform: scale(1.05);
                        box-shadow: 0 0 10px;
                        }
                    styles:
                      name:
                        - padding: 8px 1em
                        - font-family: Roboto
                        - font-size: 20px
                      card:
                        - border-radius: 30px
                        - background: "rgba(255,255,255,.1)"
                    state:
                      - value: "on"
                        styles:
                          card:
                            - background: "rgba(255,255,255,.8)"
                            - border: 1px solid white
                            - box-shadow: "0 0 10px, inset 0 0 5px"
                          name:
                            - font-weight: bold
                            - color: "rgba(0,0,0,.6)"
                          state:
                            - color: "rgba(0, 0, 0, 0.6)"
                          icon:
                            - color: "rgb(255, 178, 0)"
                            - animation: spin 2s linear infinite
                            - background: >-
                                radial-gradient(circle, rgba(255,198,0,1) 0%,
                                rgba(255,255,255,1) 54%, rgba(0,0,0,1) 100%);
                    tap_action:
                      action: toggle
                    hold_action:
                      action: more-info
`

**Reply by konradwalsh** [reply]
[outdoors-card.txt](https://github.com/dwainscheeren/dwains-lovelace-dashboard/files/5918127/outdoors-card.txt)

**Reply by konradwalsh** [reply]
This yaml card page does load this within your theme no problem:

![image](https://user-images.githubusercontent.com/7996482/106742414-ead2e180-6614-11eb-9a6d-db79262dd24f.png)

![image](https://user-images.githubusercontent.com/7996482/106742306-c8d95f00-6614-11eb-9ff9-7199fc12aeba.png)

**Reply by dwainscheeren** [reply]
Add on top
```
type: vertical-stack
cards:
```
no spaces in front of it then 2 spaces before first card/-

**Reply by konradwalsh** [reply]
Awesome, thanks
I'll post the final code for others to  use

**Reply by liveware-problems** [reply]
Presumably this should load the entire contents of the 'flip' card - with header row and buttons as detailed in outdoor-card.txt - within a custom button card directly on the room page?

If it works, that's quite a handy hack to know :-)

**Reply by dwainscheeren** [reply]
Correct

> Presumably this should load the entire contents of the 'flip' card - with header row and buttons as detailed in outdoor-card.txt - within a custom button card directly on the room page?
> 
> If it works, that's quite a handy hack to know :-)

Correct

**Reply by liveware-problems** [reply]
Is there anyway to break it out of the grid already set for the custom buttons? Or even make it double width (two buttons wide on a desktop)?

![image](https://user-images.githubusercontent.com/32192569/106772832-4d89a480-6638-11eb-8b91-8f0741828f06.png)

**Comment by dwainscheeren** [comment]
Hi,

I am closing the old Dwains Dashboard 3.x discussions as part of the DD3 cleanup.

DD3 is now in maintenance/community mode and is slowly being phased out. Active development is focused on Dwains Dashboard Next, the newest dashboard project:

https://github.com/dwainscheeren/dwains-dashboard-next

For chat, community help and discussion around DD3/DD Next, please join the Discord:

https://discord.gg/7yt64uX

If you still have a reproducible DD3 bug on the latest 3.9.x release, please open a new issue with current details. For new feature ideas, please try Dwains Dashboard Next.
